posts/2022-04-15-libertydevops-gitjenkinshelmiks.adoc

```diff
@@ -0,0 +1,180 @@
+
+       sh '''
+             docker run aquasec/trivy image $docker_login/sampleapp:v1.0
+             #eg: docker run aquasec/trivy image docker.io/someid/liberty-app:v1.0
+       '''
+
+
+== Deploying Jakarta EE & MicroProfile applications with Open Liberty to Kubernetes with Jenkins
+
+For simplicity, I will use the command line (CLI) option to configure Jenkins to deploy a Jakarta EE & MicroProfile application with Open Liberty to Kubernetes. You could use one of many tools though, such as Helm, Travis CI, Circle CI, etc.
+
+Create a new stage in the pipeline code and write all the needed commands between the shell option (sample commands give below):
+                           
+                           sh '''
+                              ibmcloud login --apikey $IBM_CLOUD_API_KEY -g $IBM_CLOUD_RSGRP
+                              ibmcloud ks cluster config --cluster $CLUSTER-ID
+                              kubectl config current-context
+                              kubectl create -f deploy/deployment.yaml #( simple k8s deployment command )
+                              kubectl create -f deploy/service.yaml #( simple k8s deployment command )
+                              kubectl create -f deploy/route.yaml #( simple k8s deployment command )
+                           '''
+
+Make sure that your Kubernetes configuration files are in the same source code repository (under a folder called "deploy"), also the image name should be updated manually or programmatically 
+
+image::/img/blog/deploymentyaml.png[Image reference in deployment yaml,width=40%,align="center"]
+
+
+When Jenkins has checked out the Java application code for the code build, all the Kubernetes configuration files are also downloaded to the Jenkins workspace so that Jenkins can run the IBM Cloud and Kubernetes commands to connect to the Kubernetes cluster and deploy the application.
+
+All the other needed application deployment commands are available in this Kubernetes command page that is straightforward 
+link:https://kubernetes.io/docs/reference/kubectl/cheatsheet/[Kubernetes sample commands].
+
+== QA testing, Jakarta EE & MicroProfile applications with Jenkins
+
+Apart from running JUnit test cases along with the code build phase, Jenkins can trigger functional and integration QA test cases automatically after deploying the Jakarta EE & MicroProfile application.
+
+Configure the test cases on jenkins job and test it manually. Create an "Authentication Token" in "Trigger builds remotely" section under "Build Triggers". Trigger this test case from docker "entrypoint" file by using remote rest API call that uses this authentication token as the identifier
+
+For example: curl -I -u <auth-token> https://<jenkins-host>/job/<job-name>/build?token=<authentication-token>
```

Comment:
Put the command into command tags (talk to Kin if you need help with tagging).

Reply:
Done